docs/rules/global-require.md

```diff
@@ -0,0 +1,58 @@
+# Disallow require() outside of the top-level module scope. (global-require)
+
+Declaring module's dependencies at the beginning of a module improves readability and provides insight to
+other developers about what modules are required. Declaring these dependencies inline within other parts of the
+code may make them harder to spot and could lead to poorly-maintainable code in the long term.
+
+In addition to maintainability issues there are also performance implications. `require` is a synchronous function and
+will block the main thread while it loads its modules. This can be a problem even after `require()` caches the loaded
+module and thus should be avoided whenever possible.
+
+
+## Rule Details
+
+This rule disallow `require()` outside of the top-level module scope.
+
+You can enable this rule with the following syntax:
+
+```json
+"global-require": 2
+```
+
+With the rule enabled enabled the following examples would warn:
+
+```
+function readFile(filename, callback) {
+    var fs = require('fs');
+    fs.readFile(filename, callback)
+}
+```
+
```

Comment:
Per comments elsewhere I assume these should be combined into a single code block